tensorflow/python/data/experimental/ops/readers.py

```diff
@@ -431,6 +431,12 @@ def make_csv_dataset_v2(
     dataset = dataset.shuffle(len(filenames), shuffle_seed)
 
   # Clean arguments; figure out column names and defaults
+  def gzip_file_io_open(filename, mode):
+    # By default, gzip will open in byte mode which will
+    # not work with csv.reader so we create a wrapper to
+    # append `t`.
+    mode = mode + "t" if "t" not in mode else mode
+    return gzip.open(filename, mode)
   if column_names is None or column_defaults is None:
     # Find out which io function to open the file
     file_io_fn = file_io.FileIO
```

Comment:
Since we only use file reading (and not writing) functionality, `file_io_fn` can be a single argument function that takes in just the filename?
`file_io_fn = lambda filename: file_io.FileIO(filename, 'r')` here and
`file_io_fn = lambda filename: gzip.GzipFile(filename, 'rt')`  below?

and in the relevant functions above:
`with file_io_fn(filename) as f:`

Reply:
Thanks @rachellim, PR has been updated.